JiraReporter/RestApiRequests.cs

```diff
@@ -21,19 +21,43 @@ public static T ResolveRequest<T>(Policy policy, RestRequest request, bool isXml
             else
                 client.Authenticator = new HttpBasicAuthenticator(policy.Username, policy.Password);
 
-            var results = client.Execute(request).Content;
+            return client;
+        }
+
+        public static T ResolveRequest<T>(Policy policy, RestRequest request, bool isXml = false)
+        {
+            var client = ClientLogin(policy);
+            var response = client.Execute(request);
+
+            ValidateResponse(response);
 
             if (isXml)
-                return Deserialization.XmlDeserialize<T>(results);
+                return Deserialization.XmlDeserialize<T>(response.Content);
             else
-                return Deserialization.JsonDeserialize<T>(results);
+                return Deserialization.JsonDeserialize<T>(response.Content);
         }
 
-        public static JiraReporter.JiraModels.Project GetProject(Policy policy)
+        public static T ResolveJiraRequest<T>(Policy policy, RestRequest request) where T : new()
+        {
+            var client = ClientLogin(policy);
+            var response = client.Execute<T>(request);
+
+            ValidateResponse(response);
+
+            return response.Data;
+        }
```

Comment:
@vladichimescu 

Pai daca ei recomanda sa se faca deserializarea manual, atunci de ce nu folosim tot timpul metoda ResolveRequest? 

Asta incercam eu sa zic - ca ar trebui ori una ori alta. Personal eu cred ca e mai bine sa folosim Execute<T> daca tot e suportat, altfel nu vad de ce ar fi acolo. Si cred ca metoda Execute<T> e si facuta sa isi dea seama automat de tipul raspunsului si sa il deserializeze fie din JSON fie din XML, deci noi nu am mai avea treaba cu asta. 

Pentru moment putem sa lasam asa, daca crezi ca exista vre-un argument, dar hai sa avem in vedere sa modificam mai incolo, daca nu modifici chiar acum. Practic nu ar trebui decat sa testam si sa vedem ca merge cu Execute<T> la fel cum mergea si pana acum cu deserializarea facuta dupa, nu?